nx/include/switch/services/ncm.h

```diff
@@ -1,120 +1,196 @@
+    NcmContentType_Program          = 1, ///< Program
+    NcmContentType_Data             = 2, ///< Data
+    NcmContentType_Control          = 3, ///< Control
+    NcmContentType_HtmlDocument     = 4, ///< HtmlDocument
+    NcmContentType_LegalInformation = 5, ///< LegalInformation
+    NcmContentType_DeltaFragment    = 6, ///< DeltaFragment
 } NcmContentType;
 
 /// ContentMetaType
 typedef enum {
-    NcmContentMetaType_SystemProgram        = 0x01,        ///< SystemProgram
-    NcmContentMetaType_SystemData           = 0x02,        ///< SystemData
-    NcmContentMetaType_SystemUpdate         = 0x03,        ///< SystemUpdate
-    NcmContentMetaType_BootImagePackage     = 0x04,        ///< BootImagePackage
-    NcmContentMetaType_BootImagePackageSafe = 0x05,        ///< BootImagePackageSafe
-    NcmContentMetaType_Application          = 0x80,        ///< Application
-    NcmContentMetaType_Patch                = 0x81,        ///< Patch
-    NcmContentMetaType_AddOnContent         = 0x82,        ///< AddOnContent
-    NcmContentMetaType_Delta                = 0x83,        ///< Delta
+    NcmContentMetaType_Unknown              = 0x0,  ///< Unknown
+    NcmContentMetaType_SystemProgram        = 0x1,  ///< SystemProgram
+    NcmContentMetaType_SystemData           = 0x2,  ///< SystemData
+    NcmContentMetaType_SystemUpdate         = 0x3,  ///< SystemUpdate
+    NcmContentMetaType_BootImagePackage     = 0x4,  ///< BootImagePackage
+    NcmContentMetaType_BootImagePackageSafe = 0x5,  ///< BootImagePackageSafe
+    NcmContentMetaType_Application          = 0x80, ///< Application
+    NcmContentMetaType_Patch                = 0x81, ///< Patch
+    NcmContentMetaType_AddOnContent         = 0x82, ///< AddOnContent
+    NcmContentMetaType_Delta                = 0x83, ///< Delta
 } NcmContentMetaType;
 
 /// ContentMetaAttribute
 typedef enum {
-    NcmContentMetaAttribute_Exfat      = (1 << 0),         ///< Exfat
-    NcmContentMetaAttribute_Rebootless = (1 << 1),         ///< Rebootless
+    NcmContentMetaAttribute_None                = 0,      ///< None
+    NcmContentMetaAttribute_IncludesExFatDriver = BIT(0), ///< IncludesExFatDriver
+    NcmContentMetaAttribute_Rebootless          = BIT(1), ///< Rebootless
 } NcmContentMetaAttribute;
 
-/// NcmContentStorage
+/// ContentInstallType
+typedef enum {
+    NcmContentInstallType_Full         = 0, ///< Full
+    NcmContentInstallType_FragmentOnly = 1, ///< FragmentOnly
+    NcmContentInstallType_Unknown      = 7, ///< Unknown
+} NcmContentInstallType;
+
+/// NcaId
 typedef struct {
-    Service s;                                             ///< IContentStorage
-} NcmContentStorage;
+    u8 c[0x10]; ///< Id
+} NcmNcaId;
 
-/// NcmContentMetaDatabase
+/// ContentMetaKey
 typedef struct {
-    Service s;                                             ///< IContentMetaDatabase
-} NcmContentMetaDatabase;
+    u64 id;                             ///< Title id.
```

Comment:
Shouldn't this be title_id?

Reply:
It’s because I based this on my NCM reimpl and Atmosphere has a specific title id type rather than using a u64. I agree though that here it’s probably more appropriate to be title_id